lineitemedit.php

```diff
@@ -161,6 +161,15 @@ function lineitemedit_civicrm_postProcess($formName, &$form) {
       }
     }
   }
+  if ('CRM_Batch_Form_Entry' == $formName) {
+    CRM_Lineitemedit_Util::disableEnablePriceField(TRUE);
+  }
+}
+
+function lineitemedit_civicrm_validateForm($formName, &$fields, &$files, &$form, &$errors) {
+  if ('CRM_Batch_Form_Entry' == $formName && empty($errors)) {
+    CRM_Lineitemedit_Util::disableEnablePriceField();
```

Comment:
What if they never successfully validate the form? The PriceFields will be left disabled.

What if others are using the system and need the price fields while someone is using the Batch Form Entry form? I don't think this is a feasible approach. Let's put more effort into dealing with the cause of the issue in that old JIRA issue.

Reply:
@JoeMurray actually I have used ```empty($errors)``` which means the PFs got disabled ONLY upon successful submission, in other words, there are no validation error messages. And then re-enabling those PFs after submission. There is no halt between these function call and there will be very short duration between these calls. 

Alternatively, the only fix I found is to remove the core function in Batch Entry.

Reply:
'Errors set by the underlying form system, e.g. for required fields, can not be unset here.' https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_validateForm/ Can you confirm that even though these types of errors cannot be unset, they are present in the $errors array?

What function do mean when you write ' the core function in Batch Entry'?

Reply:
@JoeMurray I have checked again with two required fields in the Batch entry form, set by form level (not Civi) and once the required field got values, upon submission there is no error message set in $errors array and thus $errors is empty. It is only when the code here will get triggered i.e. on successul submission. 

About the second question, the core function is [CRM_Price_BAO_PriceSet::getOnlyPriceFieldID](https://github.com/civicrm/civicrm-core/blob/master/CRM/Batch/Form/Entry.php#L463) which expects the quickconfig default price set of Contribution to have only one price-field but in our extension we are having multiple price fields for this particular price-set to support multiple quick-config line-items. So my fix here is to disable those additional price-fields on validateForm hook to bypass this exception error and reenable them on postprocess to restore them again. This is kinda hackish fix but didn't find any other alternative fix to tackle this issue unless I need to modify that core function and submit a core PR.